src/facilities-context/actions/facilities.tsx

```diff
@@ -34,17 +67,43 @@ export function selectFacility(dispatch: FacilitiesDispatch) {
 }
 
 export async function fetchFacilities(
-  scenarioId: string,
+  shouldFetchReferenceFacilities: boolean,
   dispatch: FacilitiesDispatch,
+  scenario: Scenario,
 ) {
   dispatch({ type: REQUEST_FACILITIES });
+  dispatch({ type: CLEAR_REFERENCE_FACILITIES });
   try {
-    const facilitiesList: Facilities | null = await getFacilities(scenarioId);
+    const facilitiesList: Facilities | null = await getFacilities(scenario.id);
+    let referenceFacilities: ReferenceFacilityMapping = {};
     const facilities: FacilityMapping = {};
 
     if (facilitiesList) {
+      if (shouldFetchReferenceFacilities && facilitiesList.length) {
+        // fetch reference facilities based on user facilities
+        // first facility is the reference; assume they're all the same
+        const {
+          modelInputs: { stateName },
+          systemType,
+        } = facilitiesList[0];
+        if (stateName && systemType) {
+          referenceFacilities = await fetchReferenceFacilities(
+            stateName,
+            systemType,
+            dispatch,
+          );
+        }
+      }
+
       facilitiesList.forEach((facility) => {
-        facilities[facility.id] = facility;
+        facilities[facility.id] = mergeFacilityObjects({
+          userFacility: facility,
+          referenceFacility: getReferenceFacility({
```

Comment:
Maybe I'm not understanding something, but why do we need to fetch all of the reference facilities for a given `stateName` and `systemType` and then "filter" that list down to the one that matches the facility id in the scenario's `referenceFacilitiesProp` mapping?  Couldn't we just look the reference facility's id up in this same mapping and then access it directly in firestore?

Reply:
👍